distributed/worker_state_machine.py

```diff
@@ -3082,6 +3082,10 @@ def validate_state(self) -> None:
                     ts_wait.state in READY | {"executing", "flight", "fetch", "missing"}
                     or ts_wait in self.missing_dep_flight
                     or ts_wait.who_has.issubset(self.in_flight_workers)
+                    or (
+                        ts_wait.state == "resumed"
+                        and ts_wait._previous in {"executing", "long-running"}
+                    )
```

Comment:
Without speculative assignment, it *should* not happen, but if `_previous == flight` (and consequently `_next == executing|long-running`), then ts_wait would also be in waiting_for_data.

We're also missing long-running.
Finally, lines 3083 and 3084 are redundant.